tests/components/watchyourlan/test_config_flow.py

```diff
@@ -0,0 +1,149 @@
+"""Test the WatchYourLAN config flow."""
+
+from unittest.mock import AsyncMock, patch
+
+from httpx import ConnectError
+import pytest
+
+from homeassistant.components.watchyourlan.const import DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import CONF_URL, CONF_VERIFY_SSL
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import MockConfigEntry  # Import MockConfigEntry
+
+
+@pytest.fixture(name="mock_watchyourlan_client")
+def mock_watchyourlan_client_fixture():
+    """Fixture to mock WatchYourLANClient."""
+    with patch(
+        "homeassistant.components.watchyourlan.config_flow.WatchYourLANClient",
+    ) as mock_client:
+        instance = mock_client.return_value
+        instance.get_all_hosts = AsyncMock(return_value={})
+        yield instance
```

Comment:
Move this to `conftest.py` and also patch the normal WatchYourLANClient. Check `airgradient` for an example